### ISSUE TITLE
how can we close the inappbrowser when have detect redirect to another page?

### ISSUE THREAD

**Issue by jianloong92** [proyecto26/react-native-inappbrowser#443]
how can we close the inappbrowser when have detect redirect to another page?

**Comment by krini** [comment]
on iOS you can call the close() function. On Android you cannot close it as far as I know.

**Comment by arsencenko** [comment]
@krini but how to detect specific url that inappbrowser is currently in ?

**Comment by krini** [comment]
> @krini but how to detect specific url that inappbrowser is currently in ?

You can not. If you close it, it is probably because you know the flow is done e.g. by polling a status endpoint.

**Comment by malikzype** [comment]
But that is very weired to hit a endpoint and keep checking if its successful or not. In case if the journey inapp browser is a bit long then it will be expensive operation to keep polling

**Comment by syedfaheem27** [comment]
@krini 

Is there a way to close the in app browser on android automatically. I just want to close it programmatically. Here is a code snippet.

```
      const {theme} = useTheme();

    const inAppBrowserRef = useRef(InAppBrowser);

    console.log(inAppBrowserRef, '--------------------------');

    const terminalUrls = data.endURLs;
    const initialUrl = data.initialURL;

    const navigation = useNavigation();

    useEffect(() => {
        (async () => {
            try {
                if (await inAppBrowserRef.current.isAvailable()) {
                    await InAppBrowser.openAuth(initialUrl, terminalUrls[2]);
                }
            } catch (error) {
                console.error('Error in InAppBrowser:', error);
            }
        })();

        setTimeout(() => {
            if (Platform.OS === 'android') {
                inAppBrowserRef.current.close();
                inAppBrowserRef.current.closeAuth();
            } else {
                inAppBrowserRef.current.closeAuth();
            }
            navigateBack(navigation);
        }, 4000);
    }, []);

    

```

**Comment by krini** [comment]
> [@krini](https://github.com/krini)
> 
> Is there a way to close the in app browser on android automatically. I just want to close it programmatically. Here is a code snippet.
> 
No, this line of code `inAppBrowserRef.current.close();`. will only work on ios. 

On android the browser should be closed in the browser window and not from the app.

**Comment by syedfaheem27** [comment]
@krini So, there isn't a way to close the in-app browser programmatically rather than having to close it manually.

**Comment by krini** [comment]
> [@krini](https://github.com/krini) So, there isn't a way to close the in-app browser programmatically rather than having to close it manually.

If you are in control of the page in inappbrowser you can close it from there, not from the app code on Android.

Maybe you can solve it with a deeplink back to your app - however I didn't succeed in doing this.
